chall12/estina.c

```diff
@@ -0,0 +1,78 @@
+#include <string.h>
+#include <stdio.h>
+#include <unistd.h>
+
+static void	ft_turn_parentesis(char *c)
+{
+	*c = *c == '(' ? ')' : *c == ')' ? '(' : *c;
+}
+
+static char* ft_reverse(char *str, int len)
+{
+	int		i;
+	char	c;
+
+	i = len / 2 - 1;
+	while (i >= 0)
+	{
+		ft_turn_parentesis((str + i));
+		c = str[i];
+		str[i] = str[len - i - 1];
+		str[len - i - 1] = c;
+		ft_turn_parentesis((str + i));
+		i--;
+	}
+	return (str);
+}
+
+static void	pass_flag(char c, int *flag, int *open)
+{
+	if (c == ')' && *open)
+	{
+		*open = 0;
+		(*flag)--;
+	}
+	else if (c == '(')
+	{
+		*open = 1;
+		(*flag)++;
+	}
```

Comment:
I'm not sure to get it, but it seems that if you open 2 parentheses, and close only one, open becomes `0` and you can't close any parentheses anymore?